activerecord/lib/active_record/test_fixtures.rb

```diff
@@ -39,20 +39,23 @@ def set_fixture_class(class_names = {})
       end
 
       def fixtures(*fixture_set_names)
+        env = ActiveRecord::ConnectionHandling::DEFAULT_ENV.call
+        db_config = ActiveRecord::Base.configurations.configs_for(env_name: env, spec_name: "primary")
+
         if fixture_set_names.first == :all
-          raise StandardError, "No fixture path found. Please set `#{self}.fixture_path`." if fixture_path.blank?
-          fixture_set_names = Dir["#{fixture_path}/{**,*}/*.{yml}"].uniq
-          fixture_set_names.map! { |f| f[(fixture_path.to_s.size + 1)..-5] }
+          raise StandardError, "No fixture paths found. Please set `#{self}.fixtures_paths`." if fixtures_paths.blank?
+          fixture_set_names = fixture_set_names_for(db_config)
         else
           fixture_set_names = fixture_set_names.flatten.map(&:to_s)
         end
 
         self.fixture_table_names |= fixture_set_names
-        setup_fixture_accessors(fixture_set_names)
+        setup_fixture_accessors(db_config, fixture_set_names)
       end
 
-      def setup_fixture_accessors(fixture_set_names = nil)
+      def setup_fixture_accessors(db_config, fixture_set_names = nil)
```

Comment:
I tried to rip out database configs  out of the accessors and just use paths from `fixtures_paths` but it doesn't really make sense because then we don't know what DB a fixture lives on. So, I think indexing fixture caches by DB spec names and lumping the `fixtures_paths` class option in with `primary` still makes the most logical sense.